### PR TITLE
[DPEDE-2112] Add form wrapper component first iteration

### PR DESCRIPTION
**Pull request by ASanchLumen**
https://ctl.atlassian.net/browse/DPEDE-2112

Form Wrapper component allows to dynamically create and wrap form elements. This first iteration allows to create checkboxes and radio buttons, with both inline and stacked layouts.

See index.html for usage example.

--- Also includes prettier changes

### CODE REVIEW

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import { FormWrapperCheckbox, FormWrapperRadio } from '../../constants/types';
+import { v4 as uuid4 } from 'uuid';
+import { FORM_CLASSES, UTILITY_CLASSES } from '../../constants/classes';
+
+@Component({
+  tag: 'chi-form-wrapper',
+  styleUrl: 'form-wrapper.scss',
+  scoped: true,
+})
+export class Alert {
```
Comment (xft-dev) [inline src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:17]:
```suggestion
export class FormWrapper {
```

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  @Element() el;
```
Comment (xft-dev) [inline src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:18]:
```suggestion
  @Element() el: HTMLElement;
```

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    if (!FORM_WRAPPER_LAYOUTS.includes(newValue)) {
+      throw new Error(
+        `${newValue} is not a valid type for form wrapper layout. Valid values are: ${APP_LAYOUT_FORMATS.join(' ,')}.`
+      );
+    }
+  }
+
+  @Watch('type')
+  typeValidation(newValue: string) {
+    if (!FORM_WRAPPER_LAYOUTS.includes(newValue)) {
```
Comment (xft-dev) [inline src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:55]: Shouldn't be FORM_WRAPPER_TYPES ?

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx (hunk 1 above)
Comment (ASanchLumen) [reply src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:17]: oops, solved!

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx (hunk 3 above)
Comment (ASanchLumen) [reply src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:55]: Totally, thanks!

**Review by xft-dev — APPROVED** [review]

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+  @Event() chiChange: EventEmitter<FormWrapperCheckbox[] | FormWrapperRadio[]>;
+
+  id: string = uuid4();
+
+  @Watch('layout')
+  layoutValidation(newValue: string) {
+    if (!FORM_WRAPPER_LAYOUTS.includes(newValue)) {
+      throw new Error(
+        `${newValue} is not a valid type for form wrapper layout. Valid values are: ${FORM_WRAPPER_LAYOUTS.join(' ,')}.`
```
Comment (emerson-pereira) [inline src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:48]:
```suggestion
        `${newValue} is not a valid type for form wrapper layout. Valid values are: ${FORM_WRAPPER_LAYOUTS.join(', ')}.`
```

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      );
+    }
+  }
+
+  @Watch('type')
+  typeValidation(newValue: string) {
+    if (!FORM_WRAPPER_TYPES.includes(newValue)) {
+      throw new Error(
+        `${newValue} is not a valid type for form wrapper type. Valid values are: ${FORM_WRAPPER_TYPES.join(' ,')}.`
```
Comment (emerson-pereira) [inline src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:57]:
```suggestion
        `${newValue} is not a valid type for form wrapper type. Valid values are: ${FORM_WRAPPER_TYPES.join(', ')}.`
```

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        onChiChange={(ev: CustomEvent) => this._onChange(ev)}
+      ></chi-radio-button>
+    );
+  }
+
+  _getItemWrapperClasses(isLast) {
+    return [
+      FORM_CLASSES.FORM_ITEM,
+      this.layout !== 'inline' && !isLast && UTILITY_CLASSES.MARGIN.BOTTOM[1],
+      this.layout === 'inline' && `-${this.layout}`,
```
Comment (emerson-pereira) [inline src/custom-elements/src/components/form-wrapper/form-wrapper.tsx:115]: Not sure if `-${this.layout}` should come from the const file for consistency?

**Review by emerson-pereira — APPROVED** [review]
Approved with one minor suggestion and one question.

File: src/custom-elements/src/components/form-wrapper/form-wrapper.tsx (hunk 6 above)
Comment (ASanchLumen) [reply src/custom-elements/src/components/form-wrapper/form-wrapper.tsx]: Totally agree, changed!

**Review by wojciechwlodarczyklumen — APPROVED** [review]

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Pull request merged**